Jenkinsfile

```diff
@@ -632,6 +638,72 @@ pipeline {
       }
     }
 
+    stage('Deploy Preview') {
+      when {
+        not {
+          branch 'PR-*'
+        }
+      }
+      steps {
+        setGitHubBuildStatus('nuxeo/preview', 'Deploy nuxeo Preview', 'PENDING')
+        container('maven') {
+          dir('ci/helm/preview') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute environment variables in chart values
+            sh """
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              boolean nsExist = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+              String noCommentOpt = '';
+              if (nsExist) {
+                noCommentOpt = '--no-comment'
+              } else {
+                sh "kubectl create namespace ${PREVIEW_NAMESPACE}"
+              }
+              sh "kubectl create secret generic preview-docker-cfg --namespace=${PREVIEW_NAMESPACE} --from-file=.dockerconfigjson=/home/jenkins/.docker/config.json --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"
+              // third build and deploy the chart
+              // reference branch have a different preview command. The -n name provided must be lowercase.
```

Comment:
Could you be more explicit?
Explain that for the reference branches we cannot use `jx preview` otherwise the preview deployment would be garbage collected.

Reply:
yes cf https://github.com/nuxeo/nuxeo/pull/3856#discussion_r401473144